packages/cinterop/src/jvm/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -1443,6 +1414,113 @@ actual object RealmInterop {
     }
 }
 
+/**
+ * A factory and container for various resources that can be freed when calling [free].
```

Comment:
Should we specify that this is required for any RealmValue arguments sent to the C-API?

Also, the README only mentions Strings, but the code calling this are wrapping all types?

Reply:
I will update the comments. All other datatypes are transferred by value inside the `realm_value_t` so it was an issue for strings which required external management of the buffer pointed to by `realm_value_t.string.data`. ... and also the upcoming bytearray-support in #584.